packages/amplify-graphql-relational-transformer/src/resolver/ddb-references-generator.ts

```diff
@@ -0,0 +1,297 @@
+   * @param ctx The transformer context provider.
+   */
+  makeHasManyGetItemsConnectionWithKeyResolver = (config: HasManyDirectiveConfiguration, ctx: TransformerContextProvider): void => {
+    const { field, indexName, limit, object, references, relatedType } = config;
+
+    if (references.length < 1) {
+      // TODO: Better error message
+      throw new Error('references should be populated.');
+    }
+
+    const primaryKeyFields: string[] = getPrimaryKeyFields(object);
+    const table = getTable(ctx, relatedType);
+    const dataSourceName = getModelDataSourceNameForTypeName(ctx, relatedType.name.value);
+    const dataSource = ctx.api.host.getDataSource(dataSourceName);
+    const keySchema = getKeySchema(table, indexName);
+
+    const setup: Expression[] = [
+      set(ref('limit'), ref(`util.defaultIfNull($context.args.limit, ${limit})`)),
+      ...primaryKeyFields
+        .slice(1)
+        .map((ca, idx) =>
+          set(
+            ref(`${SORT_KEY_VALUE}${idx}`),
+            methodCall(ref('util.defaultIfNull'), ref(`ctx.stash.connectionAttibutes.get("${ca}")`), ref(`ctx.source.${ca}`)),
+          ),
+        ),
+      set(ref('query'), this.makeExpression(keySchema, references)),
+    ];
+
+    // add setup filter to query
+    setup.push(
+      setArgs,
+      ifElse(
+        not(isNullOrEmpty(authFilter)),
+        compoundExpression([
+          set(ref('filter'), authFilter),
+          iff(not(isNullOrEmpty(ref('args.filter'))), set(ref('filter'), obj({ and: list([ref('filter'), ref('args.filter')]) }))),
+        ]),
+        iff(not(isNullOrEmpty(ref('args.filter'))), set(ref('filter'), ref('args.filter'))),
+      ),
+      iff(
+        not(isNullOrEmpty(ref('filter'))),
+        compoundExpression([
+          set(
+            ref('filterExpression'),
+            methodCall(ref('util.parseJson'), methodCall(ref('util.transform.toDynamoDBFilterExpression'), ref('filter'))),
+          ),
+          iff(
+            not(methodCall(ref('util.isNullOrBlank'), ref('filterExpression.expression'))),
+            compoundExpression([
+              iff(
+                equals(methodCall(ref('filterExpression.expressionValues.size')), int(0)),
+                qref(methodCall(ref('filterExpression.remove'), str('expressionValues'))),
+              ),
+              set(ref('filter'), ref('filterExpression')),
+            ]),
+          ),
+        ]),
+      ),
+    );
+
+    const queryArguments = {
+      query: raw('$util.toJson($query)'),
+      scanIndexForward: ifElse(
+        ref('context.args.sortDirection'),
+        ifElse(equals(ref('context.args.sortDirection'), str('ASC')), bool(true), bool(false)),
+        bool(true),
+      ),
+      filter: ifElse(ref('filter'), ref('util.toJson($filter)'), nul()),
+      limit: ref('limit'),
+      nextToken: ifElse(ref('context.args.nextToken'), ref('util.toJson($context.args.nextToken)'), nul()),
+    } as any;
+
+    if (indexName) {
```

Comment:
nit: IIRC `indexName` should always be defined for ref relation. This can be addressed in separate PR if it is the case.

Reply:
You're right, it should. Added to open items in the feature branch PR.